### PR TITLE
RavenDB-22264 / RavenDB-22267 / RavenDB-22272 / RavenDB-22272 More audit logs

### PR DESCRIPTION
**Pull request by arekpalinski**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-22264/Audit-log-on-specific-query-operations
https://issues.hibernatingrhinos.com/issue/RavenDB-22267/Audit-log-on-indexes-additional-sources-custom-sorters-and-analyzers
https://issues.hibernatingrhinos.com/issue/RavenDB-22272/Audit-log-on-connection-strings-operations
https://issues.hibernatingrhinos.com/issue/RavenDB-22273/Audit-log-on-changes-in-Integrations

### Additional description

Added extra logging for auditing purposes

### Type of change

- [ ] Bug fix
- [ ] Regression bug fix
- [x] Optimization
- [ ] New feature

### How risky is the change?

- [x] Low 
- [ ] Moderate 
- [ ] High
- [ ] Not relevant

### Backward compatibility

- [ ] Non breaking change
- [ ] Ensured. Please explain how has it been implemented?
- [ ] Breaking change
- [ ] Not relevant

### Is it platform specific issue?

- [ ] Yes. Please list the affected platforms.
- [x] No

### Documentation update

- [ ] This change requires a documentation update. Please mark the issue on YouTrack using `Documentation Required` tag.
- [x] No documentation update is needed 

### Testing by Contributor

- [ ] Tests have been added that prove the fix is effective or that the feature works
- [ ] Internal classes added to the test class (e.g. entity or index definition classes) have the lowest possible access modifier (preferable `private`) 
- [ ] It has been verified by manual testing

### Testing by RavenDB QA team

- [ ] This change requires a special QA testing due to possible performance or resources usage implications (CPU, memory, IO). Please mark the issue on YouTrack using `QA Required` tag.
- [ ] No special testing by RavenDB QA team is needed

### Is there any existing behavior change of other features due to this change?

- [ ] Yes. Please list the affected features/subsystems and provide appropriate explanation
- [x] No

### UI work

- [ ] It requires further work in the Studio. Please mark the issue on YouTrack using `Studio Required` tag.
- [x] No UI work is needed

### CODE REVIEW

**Review by ppekrol — APPROVED** [review]

File: src/Raven.Server/Documents/Handlers/IndexHandler.cs [hunk 1]
```diff
@@ -576,6 +576,11 @@ public async Task Reset()
         {
             var name = GetQueryStringValueAndAssertIfSingleAndNotEmpty("name");
 
+            if (LoggingSource.AuditLog.IsInfoEnabled)
+            {
+                LogAuditFor(Database.Name, $"Index {name} RESET");
```
Comment (gregolsky) [inline src/Raven.Server/Documents/Handlers/IndexHandler.cs:581]: As stated in https://issues.hibernatingrhinos.com/issue/RavenDB-22362/Unify-audit-log-lines please put the action done on a resource as the first word e.g.

RESET index {name}

File: src/Raven.Server/Documents/Handlers/Streaming/StreamingHandler.cs [hunk 2]
```diff
@@ -164,6 +166,10 @@ public async Task StreamQueryGet()
                 var ignoreLimit = GetBoolValueQueryString("ignoreLimit", required: false) ?? false;
                 var properties = GetStringValuesQueryString("field", false);
                 var propertiesArray = properties.Count == 0 ? null : properties.ToArray();
+
+                if (LoggingSource.AuditLog.IsInfoEnabled && query.Metadata.CollectionName == Constants.Documents.Collections.AllDocumentsCollection)
+                    LogAuditFor(Database.Name, $"Streaming results for all documents query: {query} (format: {format}, debug: {debug}, ignore limit: {ignoreLimit})");
```
Comment (gregolsky) [inline src/Raven.Server/Documents/Handlers/Streaming/StreamingHandler.cs:171]: How about `QUERY - streaming all documents: <the args>`

File: src/Raven.Server/Documents/Handlers/Streaming/StreamingHandler.cs [hunk 3]
```diff
@@ -247,12 +253,16 @@ public async Task StreamQueryPost()
...
 
+
                 var format = GetStringQueryString("format", false);
                 var debug = GetStringQueryString("debug", false);
                 var ignoreLimit = GetBoolValueQueryString("ignoreLimit", required: false) ?? false;
                 var properties = GetStringValuesQueryString("field", false);
                 var propertiesArray = properties.Count == 0 ? null : properties.ToArray();
 
+                if (LoggingSource.AuditLog.IsInfoEnabled && query.Metadata.CollectionName == Constants.Documents.Collections.AllDocumentsCollection)
+                    LogAuditFor(Database.Name, $"Streaming results for all documents query: {query} (format: {format}, debug: {debug}, ignore limit: {ignoreLimit})");
```
Comment (gregolsky) [inline src/Raven.Server/Documents/Handlers/Streaming/StreamingHandler.cs:264]: same thing please

File: src/Raven.Server/Integrations/PostgreSQL/Handlers/PostgreSqlIntegrationHandler.cs [hunk 4]
```diff
@@ -138,6 +139,11 @@ await using (var writer = new AsyncBlittableJsonTextWriter(context, ResponseBody
 
                     using (Database.ServerStore.ContextPool.AllocateOperationContext(out TransactionOperationContext transactionOperationContext))
                         await DatabaseConfigurations(ServerStore.ModifyPostgreSqlConfiguration, transactionOperationContext, RaftIdGenerator.DontCareId, config);
+
+                    if (LoggingSource.AuditLog.IsInfoEnabled)
+                    {
+                        LogAuditFor(Database.Name, $"Postgres integration. User {newUser.Username} PUT");
```
Comment (gregolsky) [inline src/Raven.Server/Integrations/PostgreSQL/Handlers/PostgreSqlIntegrationHandler.cs:145]: PUT user {newUser.Username} in Postgres integration

File: src/Raven.Server/Integrations/PostgreSQL/Handlers/PostgreSqlIntegrationHandler.cs [hunk 5]
```diff
@@ -192,6 +198,11 @@ await using (var writer = new AsyncBlittableJsonTextWriter(context, ResponseBody
 
                     using (Database.ServerStore.ContextPool.AllocateOperationContext(out TransactionOperationContext transactionOperationContext))
                         await DatabaseConfigurations(ServerStore.ModifyPostgreSqlConfiguration, transactionOperationContext, RaftIdGenerator.DontCareId, config);
+
+                    if (LoggingSource.AuditLog.IsInfoEnabled)
+                    {
+                        LogAuditFor(Database.Name, $"Postgres integration. User {userToDelete.Username} DELETE");
```
Comment (gregolsky) [inline src/Raven.Server/Integrations/PostgreSQL/Handlers/PostgreSqlIntegrationHandler.cs:204]: similar as above

File: src/Raven.Server/Web/System/OngoingTasksHandler.cs [hunk 6]
```diff
@@ -641,6 +642,11 @@ public async Task RemoveConnectionString()
 
             await ServerStore.EnsureNotPassiveAsync();
 
+            if (LoggingSource.AuditLog.IsInfoEnabled)
+            {
+                LogAuditFor(Database.Name, $"Connection string {connectionStringName} DELETE");
```
Comment (gregolsky) [inline src/Raven.Server/Web/System/OngoingTasksHandler.cs:647]: DELETE connection string {connectionStringName}

File: src/Raven.Server/Web/System/OngoingTasksHandler.cs [hunk 7]
```diff
@@ -787,7 +793,16 @@ private static (Dictionary<string, RavenConnectionString>, Dictionary<string, Sq
         [RavenAction("/databases/*/admin/connection-strings", "PUT", AuthorizationStatus.DatabaseAdmin)]
         public async Task PutConnectionString()
         {
-            await DatabaseConfigurations((_, databaseName, connectionString, guid) => ServerStore.PutConnectionString(_, databaseName, connectionString, guid),
+            await DatabaseConfigurations((_, databaseName, connectionString, guid) =>
+                {
+                    if (LoggingSource.AuditLog.IsInfoEnabled)
+                    {
+                        if (connectionString.TryGet(nameof(ConnectionString.Name), out string name))
+                            LogAuditFor(Database.Name, $"Connection string {name} PUT");
```
Comment (gregolsky) [inline src/Raven.Server/Web/System/OngoingTasksHandler.cs:801]: PUT...

**Pull request merged**